framework/PostProcessors/BasicStatistics.py

```diff
@@ -174,50 +172,29 @@ def inputToInternal(self, currentInp):
       return currentInput
     if currentInput.type not in ['PointSet','HistorySet']:
       self.raiseAnError(IOError, self, 'BasicStatistics postprocessor accepts PointSet and HistorySet only! Got ' + currentInput.type)
-    if currentInput.type in ['PointSet']:
-      inputDict = {'targets':{},'metadata':currentInput.getAllMetadata()}
-      for targetP in self.parameters['targets']:
-        if   targetP in currentInput.getParaKeys('input'):
-          inputDict['targets'][targetP] = currentInput.getParam('input' , targetP, nodeId = 'ending')
-        elif targetP in currentInput.getParaKeys('output'):
-          inputDict['targets'][targetP] = currentInput.getParam('output', targetP, nodeId = 'ending')
-        else:
-          self.raiseAnError(IOError, self, 'Target ' + targetP + ' has not been found in data object '+currentInput.name)
-    else:
+
+    metadata =  currentInput.getMeta(pointwise=True)
+    inputList = []
+    if currentInput.type == 'PointSet':
+      inputDict = {}
+      inputDict['targets'] = currentInput.getVarValues(self.parameters['targets'])
```

Comment:
We should improve this later to not use getVarValues, but for now it is okay.  Ideally, we can operate on the data directly without transforming it into dicts first.

Reply:
agree

Reply:
I agree, and I put a __FIXME__ for the future development.